tests comparing to numpy from_dlpack should skip on < 1.23.0

### Description

```py
@unittest.skipIf(numpy_version < (1, 22, 0), "Requires numpy 1.22 or newer")
```
is used to compare results with `np.from_dlpack` - see https://github.com/google/jax/blob/main/tests/array_interoperability_test.py#L201

[numpy added `from_dlpack` in 1.23.0](https://github.com/numpy/numpy/releases/tag/v1.23.0). In 1.22.x this was `_from_dlpack` instead.

So, when I run the tests with numpy 1.22.3, I get the error
```
E       AttributeError: module 'numpy' has no attribute 'from_dlpack'
```

Test summary:
```
FAILED tests/array_interoperability_test.py::DLPackTest::testJaxToNumpy_float32[0,4] - AttributeError: module 'nump...                                                                                                                     
FAILED tests/array_interoperability_test.py::DLPackTest::testJaxToNumpy_int8[] - AttributeError: module 'numpy' has...                                                                                                                     
FAILED tests/array_interoperability_test.py::DLPackTest::testJaxToNumpy_uint8[2,3,4] - AttributeError: module 'nump...                                                                                                                     
FAILED tests/array_interoperability_test.py::DLPackTest::testJaxToNumpy_float32[2,3,4] - AttributeError: module 'nu...                                                                                                                     
FAILED tests/array_interoperability_test.py::DLPackTest::testJaxToNumpy_uint8[4] - AttributeError: module 'numpy' h...                                                                                                                     
FAILED tests/array_interoperability_test.py::DLPackTest::testJaxToNumpy_float32[3,4] - AttributeError: module 'nump...
FAILED tests/array_interoperability_test.py::DLPackTest::testJaxToNumpy_float64[2,3,4] - AttributeError: module 'nu...                                                                                                                     
FAILED tests/array_interoperability_test.py::DLPackTest::testJaxToNumpy_int32[3,4] - AttributeError: module 'numpy'...                                                                                                                     
FAILED tests/array_interoperability_test.py::DLPackTest::testJaxToNumpy_int32[4] - AttributeError: module 'numpy' h...                                                                                                                     
FAILED tests/array_interoperability_test.py::DLPackTest::testJaxToNumpy_uint8[] - AttributeError: module 'numpy' ha...
```

### What jax/jaxlib version are you using?

jax v0.3.23, jaxlib 0.3.22

### Which accelerator(s) are you using?

CPU

### Additional system info

_No response_

### NVIDIA GPU info

_No response_

--- a/tests/array_interoperability_test.py
+++ b/tests/array_interoperability_test.py
@@ -198,7 +198,7 @@ class DLPackTest(jtu.JaxTestCase):
     shape=all_shapes,
     dtype=torch_dtypes,
   )
-  @unittest.skipIf(numpy_version < (1, 22, 0), "Requires numpy 1.22 or newer")
+  @unittest.skipIf(numpy_version < (1, 23, 0), "Requires numpy 1.23 or newer")
   @jtu.skip_on_devices("gpu")
   def testJaxToNumpy(self, shape, dtype):
     rng = jtu.rand_default(self.rng())

